build-logic/build.gradle.kts

```diff
@@ -6,8 +6,17 @@ repositories {
     gradlePluginPortal()
 }
 
+gradlePlugin {
+    plugins {
+        create("build-logic") {
+            id = "build-logic"
+            implementationClass = "org.jetbrains.BuildLogic"
+        }
+    }
+}
```

Comment:
I've had a look at the referece pull requests with the same change, and none of them create a separate plugin. 

However, I have noticed that all of them have separate build script files with the name pattern like `kotlin-jvm-convention.gradle.kts` that they then apply as a plugin in `apply(plugin = "kotlin-jvm-convention")`. 

I came to the conclusion that it's the idiomatic way of sharing build logic, and what Dokka is doing with directly referencing the code from `buildSrc` it not very idiomatic, but it needs to work as before - that's why this plugin, which imports all classes, was created. 

Is that the case?

Reply:
Yep, they depend on plugins in `.gradle.kts` but we use functions, it not the same.